### ISSUE TITLE
On Log view, the MQTT client_id is not send correctly

### ISSUE THREAD

**Issue by Zolli** [esphome/issues#1345]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
Using HomeAssistant supevisor. Home assistant is installed on a generic Ubunut 18.04 server, regularly updated

**ESP (ESP32/ESP8266, Board/Sonoff):**
Generic ESP32 wrower module

**ESPHome version (latest production, beta, dev branch)**
1.15.0-dev

**Affected component:**
MQTT Client

**Description of problem:**
* My MQTT server requires specific clientID for differrent clients, for authentication (Using VerneMQ on docker)
* The created node has a configured client_id and connects to MQTT server successfully
* When start using the log viewer on the WebUI It shows the following message:
  * INFO Connected to MQTT broker!
  * INFO Successfully reconnected to the MQTT server
* The MQTT server shows a log that tells me that the auth is not successfull
* The log contains the sent client_id, looks like this: `anon-2ywvPvaXuI4Brs7RGjHd7nP9eNs=`
* In conclusion the log viewer component is not using the configured Client_id

**Problem-relevant YAML-configuration entries:**
```yaml
mqtt:
  broker: !secret esphome.mqtt.host
  port: !secret esphome.mqtt.port
  username: !secret esphome.hp_temp_monitor.mqtt.user
  password: !secret esphome.hp_temp_monitor.mqtt.password
  client_id: !secret esphome.hp_temp_monitor.mqtt.client_id
```

**Logs (if applicable):**
```
Not applicable

```
